src/APIBasedAuth.ts

```diff
@@ -88,30 +86,111 @@ class APIBasedAuth {
         APIBasedAuth.PasswordlessAuthData
       >('/passwordless-auth', {
         appsBaseUri,
-        clientId: this.clientId,
+        clientId: this.clientOptions.clientId,
         loginAppBasePath,
         username
       })
-      .then(({ data }) => {
-        this.session = {
-          session: data.session,
-          username
-        };
+      .then(async ({ data }) => {
+        await this._store('session', { session: data.session, username });
         return data;
       })
       .catch(formatAxiosError);
   }
 
-  public logout() {
-    this.session = undefined;
+  public async logout() {
+    await this._removeFromStorage('session');
+    await this._removeFromStorage('token');
+  }
+
+  private _getStorageKeyFromType<T extends APIBasedAuth.StorageTypeName>(
+    type: T
+  ) {
+    if (type === 'session') {
+      return ['session', 'username'];
+    }
+
+    if (type === 'token') {
+      return ['accessToken', 'idToken', 'refreshToken'];
+    }
+  }
+
+  private async _getFromStorage<T extends APIBasedAuth.StorageTypeName>(
+    type: T
+  ): Promise<APIBasedAuth.StorageType<T>> | undefined {
+    if (type === 'session' && this.session) {
+      return this.session as APIBasedAuth.StorageType<T>;
+    }
+
+    if (type === 'token' && this.tokens) {
+      return this.tokens as APIBasedAuth.StorageType<T>;
+    }
+
+    const { storage, storageKeys } = this.clientOptions;
+
+    const storedValues = {} as APIBasedAuth.StorageType<T>;
+    for (const key of this._getStorageKeyFromType(type)) {
+      const value = storage.getItem(storageKeys[key] || `${type}.${key}`);
+      storedValues[key] = value instanceof Promise ? await value : value;
+    }
+
+    if (type === 'session') {
+      this.session = storedValues as APIBasedAuth.Session;
+    }
+    if (type === 'token') {
+      this.tokens = storedValues as APIBasedAuth.Tokens;
+    }
+    return storedValues;
+  }
+
+  private async _removeFromStorage<T extends APIBasedAuth.StorageTypeName>(
+    type: T
+  ) {
+    if (type === 'session') {
+      this.session = undefined;
+    }
+    if (type === 'token') {
+      this.tokens = undefined;
+    }
+
+    const { storage, storageKeys } = this.clientOptions;
+    for (const key of this._getStorageKeyFromType(type)) {
+      const value = storage.removeItem(storageKeys[key] || `${type}.${key}`);
+      if (value instanceof Promise) {
+        await value;
+      }
+    }
+  }
+
+  private async _store<T extends APIBasedAuth.StorageTypeName>(
+    type: T,
+    valuesToStore: APIBasedAuth.StorageType<T>
+  ) {
+    const { storage, storageKeys } = this.clientOptions;
+    if (type === 'session') {
+      this.session = valuesToStore as APIBasedAuth.Session;
+    }
+    if (type === 'token') {
+      this.tokens = valuesToStore as APIBasedAuth.Tokens;
+    }
+
+    for (const key of this._getStorageKeyFromType(type)) {
+      const value = storage.setItem(
+        storageKeys[key] || `${type}.${key}`,
```

Comment:
This default (`${type}.${key}`) seems pretty open, and relatively likely to collide with other keys. Can we add some sort of LifeOmic-specific prefix? In `LOAuth`, we used `lo-app-tools-auth`, for example.